tianshou/highlevel/module/actor.py

```diff
@@ -0,0 +1,261 @@
+        self.hidden_sizes = hidden_sizes
+        self.activation = activation
+
+    def create_module(self, envs: Environments, device: TDevice) -> BaseActor:
+        net_a = Net(
+            envs.get_observation_shape(),
+            hidden_sizes=self.hidden_sizes,
+            activation=self.activation,
+            device=device,
+        )
+        return continuous.Actor(
+            net_a,
+            envs.get_action_shape(),
+            hidden_sizes=(),
+            device=device,
+        ).to(device)
+
+
+class ActorFactoryContinuousGaussianNet(ActorFactoryContinuous):
+    def __init__(
+        self,
+        hidden_sizes: Sequence[int],
+        unbounded: bool = True,
+        conditioned_sigma: bool = False,
+        activation: ModuleType = nn.ReLU,
+    ):
+        """:param hidden_sizes: the sequence of hidden dimensions to use in the network structure
+        :param unbounded: whether to apply tanh activation on final logits
+        :param conditioned_sigma: if True, the standard deviation of continuous actions (sigma) is computed from the
+            input; if False, sigma is an independent parameter
+        """
+        self.hidden_sizes = hidden_sizes
+        self.unbounded = unbounded
+        self.conditioned_sigma = conditioned_sigma
+        self.activation = activation
+
+    def create_module(self, envs: Environments, device: TDevice) -> BaseActor:
+        net_a = Net(
+            envs.get_observation_shape(),
+            hidden_sizes=self.hidden_sizes,
+            activation=self.activation,
+            device=device,
+        )
+        actor = continuous.ActorProb(
+            net_a,
+            envs.get_action_shape(),
+            unbounded=self.unbounded,
+            device=device,
+            conditioned_sigma=self.conditioned_sigma,
+        ).to(device)
+
+        # init params
+        if not self.conditioned_sigma:
+            torch.nn.init.constant_(actor.sigma_param, -0.5)
+        self._init_linear(actor)
+
+        return actor
+
+
+class ActorFactoryDiscreteNet(ActorFactory):
+    def __init__(
+        self,
+        hidden_sizes: Sequence[int],
+        softmax_output: bool = True,
+        activation: ModuleType = nn.ReLU,
+    ):
+        self.hidden_sizes = hidden_sizes
+        self.softmax_output = softmax_output
+        self.activation = activation
+
+    def create_module(self, envs: Environments, device: TDevice) -> BaseActor:
+        net_a = Net(
+            envs.get_observation_shape(),
+            hidden_sizes=self.hidden_sizes,
+            activation=self.activation,
+            device=device,
+        )
+        return discrete.Actor(
+            net_a,
+            envs.get_action_shape(),
+            hidden_sizes=(),
+            device=device,
+            softmax_output=self.softmax_output,
+        ).to(device)
+
+
+class ActorFactoryTransientStorageDecorator(ActorFactory):
```

Comment:
The decorator serves to store the most recently created actor instance, such that it can be retrieved. This is necessary for the implementation of the concept of reusing the agent's preprocessing model for the critics, as we need to be able to access the object that was created by the `ActorFactory` in order to reuse it for the creation of the critics.

`IntermediateModuleFactoryFromActorFactory` is an unrelated class, which allows an `ActorFactory` to be used in contexts where an `IntermediateModuleFactory` is expected.